### PR TITLE
Unhide nodepools fixes and cleanup

### PR DESCRIPTION
**Pull request by ograu**
This was the checks, fixes and cleanups that I was doing when `unhide-nodepools` branch was accidentally merged

### CODE REVIEW

**Comment by ograu** [comment]
ping @axbarsan @oponder

File: src/components/cluster/detail/AddNodePool.js [hunk 1]
```diff
@@ -524,7 +524,7 @@ function mapStateToProps(state) {
   const { availability_zones: AZ } = state.app.info.general;
   const availabilityZones = AZ.zones;
   // More than 4 AZs is not allowed by now.
-  const maxAZ = AZ.max > 4 ? 4 : AZ.max;
+  const maxAZ = !AZ.max || AZ.max > 4 ? 4 : AZ.max;
```
[the remark this answers is not in the document]
Comment (marians) [reply src/components/cluster/detail/AddNodePool.js]: Any comment on my comment?

[the remark this answers is not in the document]
Comment (marians) [reply src/components/cluster/detail/AddNodePool.js]: Sorry, had to reload the page for comments to appear

[the remark this answers is not in the document]
Comment (marians) [reply src/components/cluster/detail/AddNodePool.js]: > So, yeah, let's enforce readability, as we have already discussed and go for the nice and readable Math.min() solution.

This didn't land in this PR though.

[the remark this answers is not in the document]
Comment (oponder) [reply src/components/cluster/detail/AddNodePool.js]: Math.min() is in! I think this is just an outdated view on the commit

[the remark this answers is not in the document]
Comment (marians) [reply src/components/cluster/detail/AddNodePool.js]: Seeing it now. Looks like the caches are strong in this one.

**Pull request merged**